src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java

```diff
@@ -226,23 +225,22 @@ public void waitAsyncTasksFinished()
 				Thread.currentThread().interrupt();
 				return;
 			}
-			if (System.currentTimeMillis() > (systemTime + executorTimeout))
+
+			if (System.currentTimeMillis() <= (systemTime + executorTimeout))
+				continue;
+
+			// If a plugin has left a runnable going and not cancelled it we could call this bad practice.
+			// We should force interrupt all these runnables, forcing them to throw Interrupted Exceptions
+			// if they handle that.
```

Comment:
I think it's a good Idea to do this, but we should give Users a way to disable this behaviour on a Test to Test Basis, since there might be cases where a long running Runnable might be needed.

Reply:
Agreed, but i think that's out of this PR's scope.

Reply:
Then this Comment and Interruption is out of scope for this PR too. I think this should be handled in it's own PR

Reply:
It's not part of this PR, it has just been moved to the left a bit.

Reply:
I see